### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by Fantom409**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,13 +22,20 @@
...
+The preferred way to install this extension is through [composer](http://getcomposer.org/download/).
+
 ```
 composer require yiisoft/rbac-php
 ```
 
+## General usage
+
+Storage stores authorization data in three PHP files specified by 
+`Storage::itemFile`, `Storage::assignmentFile`, `Storage::ruleFile`
```
Comment (viktorprogger) [inline README.md:36]:
```suggestion
`Storage::$itemFile`, `Storage::$assignmentFile`, `Storage::$ruleFile`
```
Without a `$` it looks like a constant.

File: README.md [hunk 2]
```diff
@@ -17,13 +22,20 @@
...
+
```
Comment (viktorprogger) [inline README.md:37]:
```suggestion

Php should be able to read and write these files. Non-existing files will be created automatically on any save operation.

```

File: README.md [hunk 3]
```diff
@@ -9,6 +9,11 @@
 [RBAC]: https://en.wikipedia.org/wiki/Role-based_access_control
 [Yii Framework]: https://yiiframework.com
 
+This package provides storage for [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) library. 
+It is used in Yii Framework but is supposed to be usable separately.
```
Comment (samdark) [inline README.md:13]:
```suggestion
It is used in Yii Framework but is usable separately.
```

File: README.md [hunk 4]
```diff
@@ -9,6 +9,11 @@
 [RBAC]: https://en.wikipedia.org/wiki/Role-based_access_control
 [Yii Framework]: https://yiiframework.com
 
+This package provides storage for [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) library.
```
Comment (samdark) [inline README.md:12]:
```suggestion
This package provides storage for [RBAC (Role-Based Access Control) package](https://github.com/yiisoft/rbac). 
```

File: README.md (hunk 2 above)
Comment (samdark) [reply README.md:37]: Php -> PHP

File: README.md [hunk 5]
```diff
@@ -17,13 +22,20 @@
...
 ```
 composer require yiisoft/rbac-php
 ```
 
+## General usage
+
+Storage stores authorization data in three PHP files specified by 
+`Storage::itemFile`, `Storage::assignmentFile`, `Storage::ruleFile`
+
+It is mainly suitable for authorization data that is not too big (for example, the authorization data for
```
Comment (samdark) [inline README.md:38]:
```suggestion
It is suitable for authorization data that is not too big (for example, the authorization data for
```

**Comment by samdark** [comment]
👍

**Pull request merged**